### ISSUE TITLE
Harmonize ontology identifier with Interlex

### ISSUE THREAD

**Issue by aeidi89** [HumanBrainProject/openMINDS_controlledTerms#467]
Suggestion to change "preferredOntologyIdentifier" to a list of all ontologyIdentifiers, and thus harmonize with how it is displayed in  Interlex (example link: https://neuinfo.org/interlex/view/ilx_0751626?searchTerm=alzheimer).

**Comment by apdavison** [comment]
I don't understand the suggestion. Interlex only shows a single "preferred identifier". It has a list of all identifiers, but only one of them is marked as preferred.

**Comment by aeidi89** [comment]
That is true @apdavison, the idea is to have a list of 'ontologyIdentifiers', instead of or in addition to the 'preferredOntologyIdentifier'. This came from a case where openMINDS, Interlex and Knowledge Space all referred to different preferred ontology identifiers.

**Comment by apdavison** [comment]
I agree it would be a good idea to add an array property `ontologyIdentifier` in addition to `preferredOntologyIdentifier`.

**Comment by lzehl** [comment]
Fine with me. But this issue needs to move then to the repo where the schema is defined (openMINDS_controlledTerms).

**Comment by lzehl** [comment]
TBD (@apdavison @tgbugs @UlrikeS91 @spieschnik @Raphael-Gazzotti ): 

old
-  preferredOntologyIdentifier
- InterLexIdentifier (THIS ONTOLOGY ID)
- KnowledgeSpaceLink (IRI)

new?
- ontologyIdentfier (one or list?)
- KSLink (IRI) [or is this service link?]

**Comment by UlrikeS91** [comment]
I think it is generally good to reconsider how we organize these property names. I just need to make this a bit more complicated 😉 

We need to additionally consider:
**1. SANDS atlas schemas** - They already have the property "ontologyIdentifier", which allows a single IRI. I'm not sure if these atlas elements could receive mulitple ontology IDs, but I guess they can? If it is generally the case, that an entity can have multiple ontology IDs assigned, we should always allow an array of IRIs. We can restrict it to a single one under "ontologyIdentifier" as well (instructions would specify that e.g., the preferred ID should be stated), but I do not see a got reason to mix this (i.e., allowing sometimes an array and somtimes only one IRI).

**2. ControlledTerms terminologies linking to other resources as their "preferredOntologyIdentifier"** - I just remembered this because of issue #174. There are several terminologies that link to Wikidata entries (found another example in [OpertingDevices](https://github.com/openMetadataInitiative/openMINDS_instances/blob/main/instances/latest/terminologies/operatingDevice/desktop.jsonld). Maybe we could consider these for the renaming as well? Either change the name of the property to include it somehow add a property to the ControlledTerms schemas?

Otherwise, I think it is accurate to consider KS links as service links. But I have a general question about this: Where do we draw the line for something to be considered a service? I feel like InterLex may also be considered a service. Is it not? 

Sidenote, if we decide to use ServiceLinks, the schema needs updates to fit this purpose.

**Comment by lzehl** [comment]
@openMetadataInitiative/openminds-developers 

We may want to keep a single property for "ontologyIdentifier" which is an embedded schema providing the ID and the ontology service to use to extract that term from the ontology (entry point for queries in the ontology)?

**Comment by Raphael-Gazzotti** [comment]
Although the team agreed, this is not a priority for v5.0 and has therefore been tagged as on hold.

**Comment by lzehl** [comment]
@openMetadataInitiative/openminds-developers TBD:

OPTION A:

**ontology identifier**
1. rename preferredOntologyIdentifier to ontologyIdentifier
2. change value expectation from ontologyIdentifier to array 
3. array should be regarded as ordered list with preferred ontology as first item

**non ontology identifier**
1. new property with value array (IRIs)
2. list with all other resolvable non-ontological identifiers (including KS)

OPTION B:

**ontology identifier**
1. rename preferredOntologyIdentifier to ontologyIdentifier
2. value remains one IRI (preferred ontology identifier) 

**other identifier**
1. new property with value array (IRIs)
2. list with all other resolvable ontological and non-ontological identifiers (including KS)

**Comment by tgbugs** [comment]
Suggestion:

- preferredOntologyId: single, probably contained in ontologyId as well, probably should be an owl:Class when dereferenced
- otherOntologyId: list, iris that refer to ontology classes
- database cross reference aka (xref): non-ontology entries, e.g. wiki data, ks, etc. may nonOntologyId? could do this as a pair of url + type or identifier + type frees you from having to use a url

rename these as needed

**Comment by tgbugs** [comment]
How to deal with owl:Class vs owl:NamedIndividual. What to do when there isn't really an ontology id?

**Comment by lzehl** [comment]
OLD 
- preferredOntologyIdentifier (IRI)
- InterLexIdentifier (IRI)
- KnowledgeSpaceLink (IRI)

NEW (for now) @Raphael-Gazzotti 
- preferredOntologyIdentifier (IRI)
- otherOntologyIdentifier (array of IRI)
- preferredCrossReference (IRI)
- otherCrossReference (array of IRI)

Potential future:
- preferredExternalResource 
  - (IRI) if ontologies and non-ontologies are automatically distinguishable
  - embedded schemas distinguishing ontology vs cross reference (with IRIs)
- otherExternalResource (array of IRI)
  - (IRI) if ontologies and non-ontologies are automatically distinguishable
  - embedded schemas distinguishing ontology vs cross reference (with IRIs)
